src/App.tsx

```diff
@@ -1,24 +1,225 @@
+import { getFilteredTodos } from './helpers';
+import { StatusFilter } from './types/StatusFilter';
+import { Todo } from './types/Todo';
+import { USER_ID } from './consts/consts';
+import './styles/todoapp.scss';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [statusFilter, setFilterStatus] = useState(StatusFilter.all);
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [title, setTitle] = useState('');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(errorFromServer => setError(errorFromServer.message));
+  }, []);
+
+  useEffect(() => {
+    let timerId: NodeJS.Timeout;
+
+    if (error) {
+      timerId = setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => clearTimeout(timerId);
+  }, [error]);
+
+  const todosCount = todos.length;
+  const areTodosVisible = todosCount > 0;
+
+  const visibleTodos = useMemo(() => (
+    getFilteredTodos(todos, statusFilter)
+  ), [todos, statusFilter]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter((todo) => todo.completed)
+  ), [todos, statusFilter]);
+
+  const activeTodos = useMemo(() => (
+    todos.filter((todo) => !todo.completed)
+  ), [todos, statusFilter]);
+
+  const hasSomeActiveTodos = activeTodos.length > 0;
+  const hasSomeCompletedTodos = completedTodos.length > 0;
+
+  const changeFilterStatus = useCallback((filterStaus: StatusFilter) => {
+    setFilterStatus(filterStaus);
+  }, []);
+
+  const closeNotifications = useCallback(() => setError(null), []);
+
+  const changeNotification = (errorMessage: string) => {
+    setError(errorMessage);
+  };
+
+  const changeTitle = (newTitle: string) => {
+    setTitle(newTitle);
+  };
+
+  const addTodo = useCallback((todoTitle: string) => {
+    setIsLoading(true);
+
+    const newTodo = {
+      userId: USER_ID,
+      title: todoTitle,
+      completed: false,
+    };
+
+    setTempTodo({
+      id: 0,
+      ...newTodo,
+    });
+
+    postTodo(newTodo)
+      .then((todoFromResponse) => {
+        setTodos([
+          ...todos,
+          todoFromResponse,
+        ]);
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+        setTitle('');
+      });
+  }, [todos]);
+
+  const removeTodo = useCallback((todoId: number) => {
+    setLoadingTodoIds((currentTodoIds) => [
+      ...currentTodoIds,
+      todoId,
+    ]);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos((currentTodos) => (
+          currentTodos.filter((todo) => todo.id !== todoId)
+        ));
+      })
+      .catch(() => setError('Unable to delete a todo'))
```

Comment:
In the `catch` block, you should also return an `error` from the server and show it to the user or output it to the console in case of unexpected errors:
```
.catch((error) => {
    setError(`Unable to delete a todo: ${error.message}`);
})
```